### PR TITLE
Examples: embeds, externals, and name override

### PR DESCRIPTION
**Pull request by rwaldron**
Sample output: 
https://github.com/rwaldron/johnny-five/blob/examples-embeds-externals-and-name-override/docs/led-digits-clock.md

Note that the embed does not display here, which is expected.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -183,12 +186,10 @@ module.exports = function(grunt) {
         }
 
         eg = file.read(filepath);
-        md = "docs/" + example.file.replace(".js", ".md");
-        inMarkdown = false;
+        name = (example.name || example.file).replace(".js", "");
 
-        if (!example.title) {
-          grunt.fail.fatal("No entry for " + example.file + " in titles.json");
-        }
```
Comment (rwaldron) [inline Gruntfile.js]: This is obsolete, so removed.

Comment (Resseguie) [inline Gruntfile.js]: The error message about titles.json is obsolete, but it's probably still handy to fail if they forget the required title property? Just change the error message instead?

``` js
grunt.fail.fatal("Invalid example (" + name + "): title required");
```

Comment (rwaldron) [inline Gruntfile.js]: Good idea.

File: Gruntfile.js [hunk 2]
```diff
@@ -183,11 +186,13 @@ module.exports = function(grunt) {
         }
 
         eg = file.read(filepath);
-        md = "docs/" + example.file.replace(".js", ".md");
+        name = (example.name || example.file).replace(".js", "");
```
Comment (Resseguie) [inline Gruntfile.js]: Want to change my syntax for initializing images and breadboards to match? (currently lines 224 and 228)

``` js
images = examples.images || [];
```

Comment (rwaldron) [inline Gruntfile.js]: Sure thing

**Comment by Resseguie** [comment]
I like it. A couple comments in the commits.

Since you're regenerating all the docs anyway with this, want to add a newline in .breadboards.md on line 7 after the <br>? I only put it up on the same line when I first committed the change to avoid touching every single generated example and cluttering up the diff. But if you're generating a bunch anyway, this might be a good time to clean that up, too.

File: Gruntfile.js (hunk 2 above)
Comment (rwaldron) [inline Gruntfile.js]: Same with...

```js
breadboards = example.breadboards ?
  example.breadboards :
  [{
    "name": name.replace(".js", ""),
    "title": "Breadboard for \"" + example.title + "\"",
    "auto": true
  }];
```

to...

```js
breadboards = example.breadboards || [{
    "name": name.replace(".js", ""),
    "title": "Breadboard for \"" + example.title + "\"",
    "auto": true
  }];
```

?

Comment (rwaldron) [inline Gruntfile.js]: Actually, I think I can also `name.replace(".js", "")` => `name`

**Comment by Resseguie** [comment]
One more note. In the io-plugin-eg branch I'm working on, I added a "link" property to point to a repo (or project page) for the io-plugin. I like your externals idea better. I'll merge this update in and drop the new "link" for your "externals" before doing a PR for that.

**Comment by rwaldron** [comment]
Something missing here?

**Comment by rwaldron** [comment]
>  I added a "link" property to point to a repo (or project page) for the io-plugin. I like your externals idea better.

Oh cool, sounds good.

**Comment by rwaldron** [comment]
There are links in comments in the examples that get rendered as "markdown", I'd like to eventually move those links to "externals"

**Comment by rwaldron** [comment]
I figured out the missing thing above, it was `<br>` ;)

**Comment by Resseguie** [comment]
> Something missing here?

git pull https://github.com/Resseguie/johnny-five.git examples-embeds-externals-and-name-override

(I added in the newline and regenerated docs)

**Comment by rwaldron** [comment]
Updated

**Comment by rwaldron** [comment]
Our signals crossed :\

**Comment by Resseguie** [comment]
np, you got it.

**Comment by rwaldron** [comment]
@Resseguie will wait until this is green, but good to go otherwise?

**Comment by Resseguie** [comment]
Yep, LGTM

**Comment by rwaldron** [comment]
Thanks!

**Pull request merged**